### ISSUE TITLE
require-param still wants param documentation, even if @ignore is used

### ISSUE THREAD

**Issue by paulheising** [gajus/eslint-plugin-jsdoc#780]
I use `jsdoc/recommended` which includes the `require-param` rule.

Now I have a typescript library, and I use JSDoc comments to auto-generate an html documentation for lib-users, *not for lib-devs*. Therefore, I use `@ignore` on all artifacts that are not part of the public api of my lib, so they won't appear in the generated html documentation.

However, the `require-param` rule still wants me to document params on functions that I `@ignore`d, for example:

```typescript
/** @ignore */
export function getNgModuleConfig(
  moduleClass: Record<string, unknown>,
): NgModule {
  // ...
}
```

When I run eslint, it complains:

```
64:1   warning  Missing JSDoc @param "moduleClass" declaration  jsdoc/require-param
```

From my understanding, the `@ignore` tag should take precedence over all other stuff, and therefore it should be not required to document params in `@ignore`d functions.

**Comment by brettz9** [comment]
I've filed #781 as a possible solution. It adds a new setting, `ignoreReplacesDocs`, which will be on by default, that will cause `@ignore` to replace the need for documentation.  (Some might still want to require well-formed docs even if using `@ignore`, so that's why the setting can be set to `false`, but I agree it makes sense as a default to be on.)
